From 89019a1c7c49a4c6c9f4a7ead0431fbbd8b2e931 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Tue, 5 Mar 2019 13:57:32 +0800
Subject: [PATCH 258/706] drm/imx: Don't compile i.MX ipuv3 driver as default

The patch makes i.MX ipuv3 driver don't compile as default.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit 0dbf254554d357a5079f2e3c0e3db9c9c439c74b)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/imx/Makefile       | 3 ++-
 drivers/gpu/drm/imx/imx-drm-core.c | 9 ++++++---
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/imx/Makefile b/drivers/gpu/drm/imx/Makefile
index 884f7e8f166b..a5fe34a07a20 100644
--- a/drivers/gpu/drm/imx/Makefile
+++ b/drivers/gpu/drm/imx/Makefile
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 
-imxdrm-objs := imx-drm-core.o ipuv3-crtc.o ipuv3-plane.o
+imxdrm-objs := imx-drm-core.o
 
 obj-$(CONFIG_DRM_IMX) += imxdrm.o
 
@@ -8,6 +8,7 @@ obj-$(CONFIG_DRM_IMX_PARALLEL_DISPLAY) += parallel-display.o
 obj-$(CONFIG_DRM_IMX_TVE) += imx-tve.o
 obj-$(CONFIG_DRM_IMX_LDB) += imx-ldb.o
 
+imx-ipuv3-crtc-objs  := ipuv3-crtc.o ipuv3-plane.o
 obj-$(CONFIG_DRM_IMX_IPUV3)	+= imx-ipuv3-crtc.o
 obj-$(CONFIG_DRM_IMX_HDMI) += dw_hdmi-imx.o
 obj-$(CONFIG_DRM_IMX_HDP) += hdp/
diff --git a/drivers/gpu/drm/imx/imx-drm-core.c b/drivers/gpu/drm/imx/imx-drm-core.c
index 1d053bbefc02..4d6bc960bcde 100644
--- a/drivers/gpu/drm/imx/imx-drm-core.c
+++ b/drivers/gpu/drm/imx/imx-drm-core.c
@@ -81,12 +81,12 @@ static int imx_drm_atomic_check(struct drm_device *dev,
 	ret = drm_atomic_helper_check_modeset(dev, state);
 	if (ret)
 		return ret;
-
+#ifdef arch_imx
 	/* Assign PRG/PRE channels and check if all constrains are satisfied. */
 	ret = ipu_planes_assign_pre(dev, state);
 	if (ret)
 		return ret;
-
+#endif
 	return ret;
 }
 
@@ -128,9 +128,10 @@ static void imx_drm_atomic_commit_tail(struct drm_atomic_state *state)
 	drm_atomic_helper_wait_for_flip_done(dev, state);
 
 	if (plane_disabling) {
+#ifdef arch_imx
 		for_each_old_plane_in_state(state, plane, old_plane_state, i)
 			ipu_plane_disable_deferred(plane);
-
+#endif
 	}
 
 	drm_atomic_helper_commit_hw_done(state);
@@ -409,7 +410,9 @@ static struct platform_driver imx_drm_pdrv = {
 
 static struct platform_driver * const drivers[] = {
 	&imx_drm_pdrv,
+#ifdef arch_imx
 	&ipu_drm_driver,
+#endif
 };
 
 static int __init imx_drm_init(void)
-- 
2.17.1

